src/cli.ts

```diff
@@ -15,87 +16,162 @@ const defaultCommand = define({
       type: 'string',
       default: 'auto',
       description: `Package manager to use for packing ('auto' | 'npm' | 'yarn' | 'pnpm' | 'bun')`
+    },
+    'list-tarball-files': {
+      type: 'boolean',
+      default: false,
+      description: 'List all files in the tarball',
     }
   },
   async run(ctx) {
     const root = ctx.positionals[0];
     let pack = ctx.values.pack as PackType;
+    const showAllFiles = ctx.values['list-tarball-files'] as boolean;
 
     prompts.intro('Generating report...');
 
     if (typeof pack === 'string' && !allowedPackTypes.includes(pack)) {
       prompts.cancel(
         `Invalid '--pack' option. Allowed values are: ${allowedPackTypes.join(', ')}`
       );
-      process.exit(1);
+      throw new Error('Invalid --pack option');
     }
 
     // If a path is passed, see if it's a path to a file (likely the tarball file)
+    let isTarball = false;
     if (root) {
-      const stat = await fs.stat(root).catch(() => {});
-      const isTarballFilePassed = stat?.isFile() === true;
-      if (!isTarballFilePassed) {
+      try {
+        const stat = await fs.stat(root);
+        if (stat.isFile()) {
+          const buffer = await fs.readFile(root);
+          pack = {tarball: buffer.buffer};
+          isTarball = true;
+        } else if (!stat.isDirectory()) {
+          prompts.cancel(
+            `When '--pack file' is used, a path to a tarball file must be passed.`
+          );
+          throw new Error('When --pack file is used, a path to a tarball file must be passed.');
+        }
+      } catch (error) {
         prompts.cancel(
-          `When '--pack file' is used, a path to a tarball file must be passed.`
+          `Failed to read tarball file: ${error instanceof Error ? error.message : String(error)}`
         );
-        process.exit(1);
+        throw error;
```

Comment:
why do we throw this now?

it used to exit gracefully with a non-zero exit code, but now it'll throw an uncaught exception

Reply:
yikes, i think i moved alot of stuff around while refactoring that i forgot to leave as it was.  gonna fix this.